File: core_backend/Models/DTOs/PDOKResponseDTO.cs
using Newtonsoft.Json;
using System.Text.Json.Serialization;

namespace core_backend.Models.DTOs
{
    public class PDOKResponseDTO
    {
        [JsonProperty(PropertyName = "numFound")]
        public int NumFound { get; set; }
        [JsonProperty(PropertyName = "start")]
        public int Start { get; set; }
        [JsonProperty(PropertyName = "maxScore")]
        public double MaxScore { get; set; }
        [JsonProperty(PropertyName = "numFoundExact")]
        public bool NumFoundExact { get; set; }
        [JsonProperty(PropertyName = "docs")]
        public List<PDOKLocationDTO> Docs { get; set; }
    }
}

File: desktop/lib/infastructure/repositories/dtos/pdok_response_dto.dart
import 'package:desktop/infastructure/repositories/dtos/pdok_location_dto.dart';

class PDOKResponseDTO {
  double maxScore;
  int numFound, start;
  bool numFoundExact;
  List<PDOKLocationDTO> locations;

  PDOKResponseDTO({required this.maxScore, required this.numFound, required this.start, required this.numFoundExact, required this.locations});

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['Lat'] = maxScore;

    return data;
  }

  static List<PDOKResponseDTO> fromList(List<dynamic> list) {
    List<PDOKResponseDTO> mappedList = [];
    for (var item in list) {
      mappedList.add(PDOKResponseDTO.fromMap(item));
    }

    return mappedList;
  }

  factory PDOKResponseDTO.fromMap(Map<String, dynamic> json) => PDOKResponseDTO(
      maxScore: json["maxScore"] / 1.0,
      numFound: json["numFound"],
      start: json["start"],
      numFoundExact: json["numFoundExact"],
      locations: json["docs"] != null ? PDOKLocationDTO.fromList(json["docs"]) : []);
}
